### ISSUE TITLE
PNG files not transparent

### ISSUE THREAD

**Issue by numerique-mood** [gjb2048/moodle-format_grid#142]
Moodle : 3.10.3
Format Grid : 3.10.1.1
Theme : Moove

Hi, 
It seems that our last "Format Grid" plugin update, png files in the grid have a black background instead of transparency.
It only appears in the grid view not in the activities (screenshot below)
I try to change the theme to Boost or Classic but no changes.
Do you know this problem ?
Thanks for your work,
Sébastien

![Capture d’écran 2021-05-18 à 15 38 33](https://user-images.githubusercontent.com/84393096/118662084-0478ed00-b7f0-11eb-9c6e-6dfa62986f23.png)

**Comment by gjb2048** [comment]
Cannot replicate, please upload a course backup without users (an MBZ).

**Comment by numerique-mood** [comment]
Thanks for the reply.
See below for the backup.
Regards,
Sébastien

[sauvegarde-moodle2-course-251-support_plateforme_enseignants-20210518-1557-nu.mbz.zip](https://github.com/gjb2048/moodle-format_grid/files/6501859/sauvegarde-moodle2-course-251-support_plateforme_enseignants-20210518-1557-nu.mbz.zip)

**Comment by gjb2048** [comment]
Thanks I can now replicate, don't know why though.

**Comment by numerique-mood** [comment]
I looked in the recently modifications of my Moodle but, the update is the last. Just before I added modifications in scss of Moove but when I erase them it doesn't change. 
I will try on a "fresh" install with only Moove and Grid View.

**Comment by gjb2048** [comment]
Ok, something to do with the alpha channel of the PHP imagepng processing -> https://www.php.net/manual/en/function.imagesavealpha.php - tried a few ideas, no joy, don't know how to solve at the moment.

Please use a workaround by setting the background of the PNG to the colour you need.

**Comment by gjb2048** [comment]
Note:  Previous code did this by default using the setting contained in the format / course.

**Comment by gjb2048** [comment]
@numerique-mood Ok, I think I have a solution.  I've also been refactoring things, so please patch your 'lib.php' 'generate_image' function with these specific additions https://github.com/gjb2048/moodle-format_grid/commit/3cda0333640061e8661c0e0441a49f57f12641c3#diff-33552c0b997f7b199f66c9bb89baeef2047b889c15ce7afe5501c5af864ec30eL1061-L1110 - then change something, like the ratio or the size so that the displayed images are regenerated.  Take a backup of lib.php first!

**Comment by numerique-mood** [comment]
OK, I will try this. For information, on another Moodle installation where I didn't update Grid Format (3.10.1.0), I downloaded the old version, delete 3.10.1.1 and reupload the old version and it works.

**Comment by gjb2048** [comment]
@numerique-mood Thanks and indeed, try on a test server!  And the reason between 3.10.1.0 and 3.10.1.1 is -> https://github.com/gjb2048/moodle-format_grid/issues/142#issuecomment-843239943

**Comment by numerique-mood** [comment]
> @numerique-mood Ok, I think I have a solution. I've also been refactoring things, so please patch your 'lib.php' 'generate_image' function with these specific additions [3cda033#diff-33552c0b997f7b199f66c9bb89baeef2047b889c15ce7afe5501c5af864ec30eL1061-L1110](https://github.com/gjb2048/moodle-format_grid/commit/3cda0333640061e8661c0e0441a49f57f12641c3#diff-33552c0b997f7b199f66c9bb89baeef2047b889c15ce7afe5501c5af864ec30eL1061-L1110) - then change something, like the ratio or the size so that the displayed images are regenerated. Take a backup of lib.php first!

Sorry, I find the lib.php but I don't know where I ave to add those additions... I'm not a programer 😕

**Comment by gjb2048** [comment]
@numerique-mood Ok, make a backup of your current lib.php and replace it with this one https://raw.githubusercontent.com/gjb2048/moodle-format_grid/master_142/lib.php.

**Comment by gjb2048** [comment]
@numerique-mood In that case I don't know as works for me.  Is there anything more in the PHP error log?

**Comment by numerique-mood** [comment]
The error seems to apply on all the Moodle and not only in the curse...
![Capture d’écran 2021-05-18 à 18 04 15](https://user-images.githubusercontent.com/84393096/118685827-b7067b00-b803-11eb-8d9b-f57beb503992.png)

Just to be shure, it's the lib.php of your plugin ?

**Comment by gjb2048** [comment]
Yes!  Please put the old one back.  I don't know why its happening.

**Comment by gjb2048** [comment]
Unless you've downloaded the new one and saved it incorrectly.

**Comment by numerique-mood** [comment]
Sorry my mistake... I put the old one back and I change my process : I replace directly all the lines of codes on my server (FTP) with the newest, save it and then I refresh, the error disappears ! And when I change the ratio of the tiles as you mentioned before, the black background disappears too !

**Comment by numerique-mood** [comment]
Everything is OK ! I just made the same process on my Moodle school and it works ! Thanks again for your help !

**Comment by gjb2048** [comment]
@numerique-mood Thanks, I'll close it after I've backported / released etc.
